src/main.ts

```diff
@@ -53,37 +34,98 @@ const command: Command & { __runner: any } = {
 			type: 'boolean',
 			default: false
 		});
-	},
-	run: async (helper: Helper, args: { pattern: string; dry: boolean }) => {
+	}
+
+	async getConfigs(fromVersion: number, toVersion: number = LATEST_VERSION): Promise<VersionConfig[]> {
+		const configs: VersionConfig[] = [];
+		for (let i = fromVersion + 1; i <= toVersion; ++i) {
+			configs.push((await import(`./v${i}/main`)).config as VersionConfig);
+		}
+		return configs;
+	}
+
+	run = async (helper: Helper, args: { pattern: string; dry: boolean }) => {
 		const { pattern, dry } = args;
 		const paths = glob.sync(pattern);
 		const hasJSX = paths.some((p: string) => p.match(/\.tsx$/g));
-		const opts = {
-			parser: hasJSX ? 'typescript-jsx' : 'typescript',
-			transform: path.resolve(__dirname, 'transforms', 'module-transform-to-framework.js'),
-			path: paths,
-			verbose: 0,
-			babel: false,
-			dry,
-			extensions: 'js',
-			runInBand: false,
-			silent: false
-		};
+		const parser = hasJSX ? 'typescript-jsx' : 'typescript';
+		const fromVersion = await this.depManager.getDojoVersion();
+		const toVersion = LATEST_VERSION;
+
+		if (toVersion <= fromVersion) {
+			throw Error(`Attempt to upgrade from ${fromVersion} to ${toVersion} not allowed. Exiting.`);
+		}
+
 		if (!dry) {
-			const answer = await inquirer.prompt({
+			const answer = await prompt({
 				type: 'confirm',
 				name: 'run',
 				message:
 					'This command will irreversibly modify files. Are you sure you want to run the upgrade? Use the --dry option first if in doubt',
 				default: false
 			});
-			if (!(answer as any).run) {
-				throw Error('Aborting upgrade');
+
+			if (!answer) {
+				throw Error('Aborting upgrade.');
 			}
-			return await run(opts);
 		}
-		return await run(opts);
+
+		await Promise.all(
+			(await this.getConfigs(fromVersion, toVersion)).map((config) => this.runUpgrade(config, parser, paths, dry))
```

Comment:
won't this fail if it tries to run `npm i @dojo/framework@3.x` and `npm i @dojo/framework@4.x` at the same time?

Reply:
This seems like a legitimate concern, each version upgrade should be don't incrementally.